apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/server/S3DirectoryDao.java

```diff
@@ -406,53 +413,73 @@ public InputStream openStream() throws IOException {
 
   /**
-   * Read {@link ObjectMetadata} for the given S3 key. Recognize the possible case of object not
-   * found (HTTP 404) by throwing more useful {@link FileNotFoundException}.
+   * Read {@link HeadObjectResponse} metadata for the given S3 key. Recognize the possible case of
+   * object not found (HTTP 404) by throwing more useful {@link FileNotFoundException}.
    *
    * @param fileName the simple file name for the object
    * @param s3Key the S3 object key
    * @return the meta data
-   * @throws IOException eith {@link FileNotFoundException} or an AWS runtime exception
+   * @throws IOException with {@link FileNotFoundException} or an AWS runtime exception
    */
-  private ObjectMetadata readS3ObjectMetaData(String fileName, String s3Key) throws IOException {
+  private HeadObjectResponse readS3ObjectMetaData(String fileName, String s3Key)
+      throws IOException {
     try {
-      return s3Client.getObjectMetadata(s3BucketName, s3Key);
-    } catch (AmazonS3Exception ex) {
-      if (ex.getStatusCode() == AWS_NOT_FOUND_STATUS_CODE) {
-        var fileNotFound = new FileNotFoundException(fileName);
-        fileNotFound.addSuppressed(ex);
-        throw fileNotFound;
-      } else {
-        throw ex;
-      }
+      HeadObjectRequest headObjectRequest =
+          HeadObjectRequest.builder().bucket(s3BucketName).key(s3Key).build();
+      return s3Client.headObject(headObjectRequest);
+    } catch (NoSuchKeyException | NoSuchBucketException e) {
+      var fileNotFound = new FileNotFoundException(fileName);
+      fileNotFound.addSuppressed(e);
+      throw fileNotFound;
     }
   }
 
   /**
-   * Download S3 object and return its {@link ObjectMetadata}. Recognize the possible case of object
-   * not found (HTTP 404) by throwing more useful {@link FileNotFoundException}.
+   * Download S3 object and return its {@link GetObjectResponse}. Recognize the possible case of
+   * object not found (HTTP 404) by throwing more useful {@link FileNotFoundException}.
    *
-   * @param fileName the simple file name for the object
    * @param s3Key the S3 object key
    * @param tempDataFile where to store the downloaded object
    * @return the meta data
-   * @throws IOException eith {@link FileNotFoundException} or an AWS runtime exception
+   * @throws IOException with {@link FileNotFoundException} or an AWS runtime exception
    */
-  private ObjectMetadata downloadS3Object(String fileName, String s3Key, Path tempDataFile)
-      throws IOException {
+  private GetObjectResponse downloadS3Object(String s3Key, Path tempDataFile)
+      throws FileNotFoundException {
+    /* Gather information needed to prepare the S3TransferManager */
+    String bucketLocation =
+        s3Client
+            .getBucketLocation(GetBucketLocationRequest.builder().bucket(s3BucketName).build())
+            .locationConstraintAsString();
+    // bucketLocation is null if the location is us-east-1, so we need to handle that result
+    // specifically
```

Comment:
Is it only us-east-1 that has null for bucketLocation? I know we chatted once about being able to flip to west for fallover purposes, so wanted to make sure this wasnt hardcoded to east if it would happen in other configurations.

Reply:
Correct, it's `locationConstraintAsString` specifically which returns null when the region is us-east-1. I'll update the comment to make this clearer